packages/terra-date-picker/src/DateUtil.js

```diff
@@ -1,22 +1,33 @@
 /* eslint-disable no-underscore-dangle */
-import moment from 'moment';
+import moment from 'moment-timezone';
 
 class DateUtil {
   /**
    * Creates a moment object using the provided date string. Moment is unable to initialize a valid date if the date passed in is
    * null, empty string, or alpha characters and undefined would be returned.
    * @param {string|undefined} date - The date to convert. Expect to be in ISO format.
+   * @param {string|undefined} initialTimeZone - The timezone of the date.
    * @return {object|undefined} - The moment object. Undefined if unable to convert.
    */
-  static createSafeDate(date) {
+  static createSafeDate(date, initialTimeZone) {
     if (!date) {
       return undefined;
     }
 
-    let momentDate = moment(date, DateUtil.ISO_EXTENDED_DATE_FORMAT, true);
+    let momentDate;
+    if (initialTimeZone && moment.tz.zone(initialTimeZone)) {
+      momentDate = moment.tz(date, DateUtil.ISO_EXTENDED_DATE_FORMAT, true, initialTimeZone);
+    } else {
+      momentDate = moment(date, DateUtil.ISO_EXTENDED_DATE_FORMAT, true);
+    }
+
     if (!momentDate || !momentDate.isValid()) {
       // This should allow DateTime inputs that used to work in the moment.ISO_8601 to still pass but discard the time from the value.
-      momentDate = moment(date.slice(0, 10), DateUtil.ISO_EXTENDED_DATE_FORMAT, true);
+      if (initialTimeZone && moment.tz.zone(initialTimeZone)) {
```

Comment:
nit: You can store `initialTimeZone && moment.tz.zone(initialTimeZone` in a const and use it in both places. Not sure how expensive it is to make this check.

Reply:
Changed in this commit a63c5545ec2835f9c76c1642e14c9902b50c67eb